Do not show window title and replace some deprecated X11 functions.
You have to manually remove all "title" members in config.h (Rule structure)
--- a/dwm.c	2011-12-19 17:02:46.000000000 +0200
+++ b/dwm.c	2015-02-23 19:49:50.305463478 +0200
@@ -36,6 +36,7 @@
 #include <X11/Xlib.h>
 #include <X11/Xproto.h>
 #include <X11/Xutil.h>
+#include <X11/XKBlib.h>
 #ifdef XINERAMA
 #include <X11/extensions/Xinerama.h>
 #endif /* XINERAMA */
@@ -148,7 +149,7 @@
 typedef struct {
 	const char *class;
 	const char *instance;
-	const char *title;
+	/*const char *title;*/
 	unsigned int tags;
 	Bool isfloating;
 	int monitor;
@@ -242,7 +243,7 @@
 static void updatesizehints(Client *c);
 static void updatestatus(void);
 static void updatewindowtype(Client *c);
-static void updatetitle(Client *c);
+/*static void updatetitle(Client *c);*/
 static void updatewmhints(Client *c);
 static void view(const Arg *arg);
 static Client *wintoclient(Window w);
@@ -307,8 +308,8 @@
 
 	for(i = 0; i < LENGTH(rules); i++) {
 		r = &rules[i];
-		if((!r->title || strstr(c->name, r->title))
-		&& (!r->class || strstr(class, r->class))
+		/*if((!r->title || strstr(c->name, r->title))*/
+		if ((!r->class || strstr(class, r->class))
 		&& (!r->instance || strstr(instance, r->instance)))
 		{
 			c->isfloating = r->isfloating;
@@ -1084,7 +1085,7 @@
 	XKeyEvent *ev;
 
 	ev = &e->xkey;
-	keysym = XKeycodeToKeysym(dpy, (KeyCode)ev->keycode, 0);
+	keysym = XkbKeycodeToKeysym(dpy, (KeyCode)ev->keycode, 0, 0);
 	for(i = 0; i < LENGTH(keys); i++)
 		if(keysym == keys[i].keysym
 		&& CLEANMASK(keys[i].mod) == CLEANMASK(ev->state)
@@ -1116,7 +1117,7 @@
 	if(!(c = calloc(1, sizeof(Client))))
 		die("fatal: could not malloc() %u bytes\n", sizeof(Client));
 	c->win = w;
-	updatetitle(c);
+	/*updatetitle(c);*/
 	if(XGetTransientForHint(dpy, w, &trans) && (t = wintoclient(trans))) {
 		c->mon = t->mon;
 		c->tags = t->tags;
@@ -1314,7 +1315,7 @@
 			break;
 		}
 		if(ev->atom == XA_WM_NAME || ev->atom == netatom[NetWMName]) {
-			updatetitle(c);
+			/*updatetitle(c);*/
 			if(c == c->mon->sel)
 				drawbar(c->mon);
 		}
@@ -1995,13 +1996,14 @@
 	             && c->maxw == c->minw && c->maxh == c->minh);
 }
 
-void
+/* hack to mark broken clients */
+/*void
 updatetitle(Client *c) {
 	if(!gettextprop(c->win, netatom[NetWMName], c->name, sizeof c->name))
 		gettextprop(c->win, XA_WM_NAME, c->name, sizeof c->name);
-	if(c->name[0] == '\0') /* hack to mark broken clients */
+	if(c->name[0] == '\0') 
 		strcpy(c->name, broken);
-}
+}*/
 
 void
 updatestatus(void) {
